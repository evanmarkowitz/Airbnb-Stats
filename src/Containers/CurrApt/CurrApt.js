import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

export const CurrApt = ({currApt}) => {

  return(
    <article className='curr-apt'>
      <h2>Selected Apartment</h2>
      <p><span className='key'>Name:</span> {currApt.name}</p>
      <p><span className='key'>Price:</span> {currApt.price}</p>
      <p><span className='key'>Number of Reviews:</span> {currApt.numReviews}</p>
      <p><span className='key'>Link:</span> <a href={`www.airbnb.com/rooms/${currApt.id}`}>AirBNB</a></p>
    </article>
  )
}

CurrApt.propTypes = {
  currApt: PropTypes.string
}
export const mapStateToProps = (state) => ({
  currApt: state.currApt
})

export default connect(mapStateToProps)(CurrApt)