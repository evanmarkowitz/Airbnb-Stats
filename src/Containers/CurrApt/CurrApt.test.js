import React from 'react';
import { CurrApt, mapStateToProps } from './CurrApt';
import {shallow} from 'enzyme'


describe('Current Apartment', ()=> {
    let wrapper;
    let currApt
  beforeEach(() => {
    currApt = {aptType: "Entire home/apt",
      hood: "Long Island City",
      id: 21530562,
      lat: 40.75358595672167,
      long: -73.93849189432474,
      name: "Modern Cozy & Clean Private 3 Bedroom Mins to NYC",
      numReviews: 60,
      price: 300, 
      reviewsPerMonth:3.21}
    wrapper = shallow(<CurrApt currApt={currApt}/>)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should map dispatch to props', () => {
    let props= {currApt: currApt}
    const mappedProps = mapStateToProps(props);
    expect(mappedProps).toEqual(props);
  })
})