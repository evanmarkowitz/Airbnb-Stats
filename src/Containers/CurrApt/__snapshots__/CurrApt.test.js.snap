// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current Apartment should match the snapshot 1`] = `
<CurrApt
  currApt={
    Object {
      "aptType": "Entire home/apt",
      "hood": "Long Island City",
      "id": 21530562,
      "lat": 40.75358595672167,
      "long": -73.93849189432474,
      "name": "Modern Cozy & Clean Private 3 Bedroom Mins to NYC",
      "numReviews": 60,
      "price": 300,
      "reviewsPerMonth": 3.21,
    }
  }
/>
`;
